subprojects/core/src/main/java/org/gradle/internal/classpath/InstrumentingClasspathFileTransformer.java

```diff
@@ -221,19 +273,55 @@ public static Policy instrumentForLoadingWithAgent() {
         return new Policy() {
             @Override
             public void applyConfigurationTo(Hasher hasher) {
-                hasher.putInt(1);
+                hasher.putInt(AGENT_INSTRUMENTATION_VERSION);
             }
 
             @Override
             public boolean instrumentFile(File file) {
                 return true;
             }
 
+            @Override
+            public byte[] processManifest(File source, ClasspathEntryVisitor.Entry entry) throws IOException {
+                try {
+                    Manifest parsedManifest = new Manifest(new ByteArrayInputStream(entry.getContent()));
+                    if (Boolean.parseBoolean(parsedManifest.getMainAttributes().getValue(MULTI_RELEASE_ATTRIBUTE))) {
```

Comment:
♻️ Extract complex expression to its own method: `if (isMultiReleaseJarManifest(parsedManifest)) {`